File: NtgeCore.Net/src/X25519/X25519PrivateKey.cs
using System;

namespace NtgeCore.Net.X25519
{
    public class X25519PrivateKey : RustObject
    {
        internal X25519PrivateKey(IntPtr ptr) : base(ptr)
        {
        }

        public override void Dispose()
        {
            Native.destroyX25519PrivateKey(Ptr);
        }
    }

}

File: NtgeCore-Android/ntge/src/main/java/com/dimension/ntge/x25519/X25519PrivateKey.kt
package com.dimension.ntge.x25519

import com.dimension.ntge.Ntge

class X25519PrivateKey internal constructor(
        internal val ptr: Long
) : AutoCloseable {

    override fun close() {
        Ntge.destroyX25519PrivateKey(ptr)
    }
}

